chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/RoomStore.kt

```diff
@@ -1,164 +1,118 @@
+                        (new.userIds - existing.userIds).map { UserInternalEvent.UserJoinedRoom(it, new.roomId) } +
+                        (existing.userIds - new.userIds).map { UserInternalEvent.UserLeftRoom(it, new.roomId) }
                     }
+                }
+                is UserSubscriptionEvent.AddedToRoomEvent -> {
+                    this.rooms[event.room.id] = event.room
+                    this.members[event.memberships.roomId] = event.memberships
+                    this.readStates[event.readState.roomId] = event.readState
 
-                    listOf(event) + addedTo + removedFrom + updated + usersJoined + usersLeft
+                    listOf(UserInternalEvent.AddedToRoom(this[event.room.id]!!))
                 }
-                is UserSubscriptionEvent.AddedToRoomEvent ->
-                    listOf(event.also {
-                        this += event.room
-                    })
                 is UserSubscriptionEvent.RoomUpdatedEvent -> {
-                    var newRoom = event.room
-                    val knownRoom = roomsMap[newRoom.id]!!
-                    // we don't get unread count and members with this event
-                    newRoom = newRoom.copy(memberUserIds = knownRoom.memberUserIds)
-                    if (knownRoom.unreadCount != null) {
-                        // we only get unread counts in initial state for 1000 rooms
-                        newRoom = newRoom.copy(unreadCount = knownRoom.unreadCount)
-                    }
-                    this += newRoom
-                    listOf(UserSubscriptionEvent.RoomUpdatedEvent(newRoom))
+                    this.rooms[event.room.id] = event.room
+                    listOf(UserInternalEvent.RoomUpdated(this[event.room.id]!!))
                 }
                 is UserSubscriptionEvent.ReadStateUpdatedEvent -> {
-                    var room = roomsMap[event.readState.roomId]!!
-                    room = room.copy(unreadCount = event.readState.unreadCount)
-                    this += room
-
-                    // Returning a copy with no cursor so that the returned applied event is
-                    // not translated to NewReadCursor but to RoomUpdated ChatEvent.
-                    // That is the relevant event from the perspective of this store.
-                    // The ChatEvent.NewReadCursor will be translated
-                    // into if CursorStore updates applying ReadStateUpdatedEvent and
-                    // returning it with the cursor.
-                    val eventWithNoCursor = event.copy(event.readState.copy(cursor = null))
-                    listOf(eventWithNoCursor)
+                    this.readStates[event.readState.roomId] = event.readState
+                    listOf(
+                            UserInternalEvent.RoomUpdated(this[event.readState.roomId]!!),
+                            UserInternalEvent.NewCursor(event.readState.cursor!!) // can this be null?
```

Comment:
I think it can. A cursor can be not set (yet) for the current user, but new messages are being sent by other users (unread count increments). We should only return `NewCursor` event if there was one that came with the event.

Reply:
This has been resolved (the room store should have nothing to do with cursors, and it no longer does)